### PR TITLE
Add the alphametics exercise & babel-polyfil dependency

### PR DESCRIPTION
**Pull request by arthurchipdean**

### CODE REVIEW

**Comment by arthurchipdean** [comment]
Why does it keep including all the other commits that have already been merged? Also looks like the build failed again.

**Comment by matthewmorgan** [comment]
@arthurchipdean I'm not sure why you're seeing those commits, but that's not a big deal-- you can squash everything as we discussed before.

I'm guessing your build failure is because you have a/some `package.json` files in individual exercises that don't match the global one.  Running `make test` should copy the global file to each directory.  I would give that a try, commit the result, then push.  See if that resolves the build issue.

**Comment by arthurchipdean** [comment]
@matthewmorgan I tried that already and it wasn't working. It's listing all files from those commits as new after I run git reset --soft <hash>. I just committed the result.

**Comment by matthewmorgan** [comment]
OK, I will pull down this branch and have a closer look.  I'll let you know what I find out.  Thanks for your efforts!

File: exercises/alphametics/jest_0/haste-map-b9c6a92470dac2bfa148c7d6f9b7f8fb-7de207d7447bebf9b1fa9b3dc26b900a [hunk 1]
```diff
@@ -0,0 +1 @@
+{"clocks":{},"files":{"/root/xecmascript/exercises/alphametics/alphametics.spec.js":["",1495674576000,1,["./example"]],"/root/xecmascript/exercises/alphametics/example.js":["",1495674304000,1,[]]},"map":{},"mocks":{}}
```
Comment (tejasbubane) [inline exercises/alphametics/jest_0/haste-map-b9c6a92470dac2bfa148c7d6f9b7f8fb-7de207d7447bebf9b1fa9b3dc26b900a]: please remove this file

**Comment by matthewmorgan** [comment]
OK,

A couple of things:

1. When I tried using `reset --soft` it appeared to work.  I can't be sure, because I can't push back to your branch.  FYI I ran `git reset --soft 5252b42`, then committed the result, and got only one commit in my history.  GItHub may be doing something different, but it worked with git.
2.  @tejasbubane called out an issue with the Makefile in #301 .  I verified that problem, and when I commented out the `test-package-files` command, I was able to run the test suite on your branch with no problem.

Bottom line, I think we need to fix #301 before we can build anything, including your branch.  If I have time today I will check it out....I think it may be as simple as setting the `-max-depth` parameter on `find`.

**Comment by tejasbubane** [comment]
@arthurchipdean Can you check if this works? https://babeljs.io/docs/plugins/transform-regenerator/#usage-via-babelrc-recommended-

Since babel-plugin-transform-regenerator is already installed, I think it would be better if that can be used instead of adding a new package.

**Comment by arthurchipdean** [comment]
@tejasbubane  Yes that worked! I committed the change.

**Comment by arthurchipdean** [comment]
@matthewmorgan I tried that git reset soft using the hash you provided and it seems to be adding more commits.

**Comment by tejasbubane** [comment]
@arthurchipdean We have Babel settings in package.json ( see babel
section). Add the transform plugin there and remove the .babelrc file.

On Jun 2, 2017 8:19 PM, "Arthur Dean" <notifications@github.com> wrote:

@matthewmorgan <https://github.com/matthewmorgan> I tried that git reset
soft using the hash you provided and it seems to be adding more commits.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/exercism/xecmascript/pull/299#issuecomment-305810302>,
or mute the thread

**Comment by matthewmorgan** [comment]
@arthurchipdean I pulled your branch and it looks like you are in the middle of a `REBASE` that has created some merge conflicts.  This is very common, and should be resolvable.  I would suggest following these steps:

1. Edit the `config.json` file to remove the `problems` key and its values.  Please also remove any conflict artifacts like `==========` etc.
2. `git add config.json`
3. `git rebase --continue`

I believe this should put you back in a proper state.  Then try pushing the changes back up and let's see what we have.  If that doesn't work, don't worry- I'll help you sort it out!

**Comment by arthurchipdean** [comment]
@matthewmorgan I followed your steps.

**Comment by matthewmorgan** [comment]
@arthurchipdean the current problem is that the `package.json` file in one or more exercises does not match the global `package.json` file, so the `test-package-files` task in the makefile is failing.

Unfortunately, there is not a simple way to copy that file to every exercise.  That is our fault-- we prematurely merged that new task into the makefile without providing an easy way to copy those files over.  Let me see what I can do about that.

**Comment by matthewmorgan** [comment]
@arthurchipdean I've made a change to `master` that will work around this problem for now.  You can either pull that change in to your branch, or simply delete line 36 from the Makefile:

```bash
	@$(MAKE) -s test-package-files
```

 and re-run `make test` from the `xecmascript/exercises` directory.  This should copy all those files over to the individual directories for you.

I'll open an issue to modify the Makefile to perform this task automatically so others don't have the same issue.  I feel terrible that this has been such a hassle for you.

**Comment by matthewmorgan** [comment]
@arthurchipdean is there anything I can do do help you get this one finished?

**Comment by arthurchipdean** [comment]
@matthewmorgan sorry for the delay, I have been busy. I'm attempting your feedback right now. Tests are running. So after running the tests I'll just commit the result and push?

**Comment by arthurchipdean** [comment]
It's failing here :(
```
 FAIL  exercises/connect/tmp/temp.connect.spec.js
  ● Test suite failed to run

    Cannot find module './_copyObject' (While processing preset: "/root/xecmascript/exercises/connect/node_modules/babel-preset-env/lib/index.js")

```

**Comment by matthewmorgan** [comment]
@arthurchipdean no worries I have been busy myself!  End of school year etc.  

I pulled down your branch and was able to run it locally with no trouble at all.  I'm going to try making another branch with your code and opening a PR to see if there's any issue in CI for me.

**Comment by matthewmorgan** [comment]
@arthurchipdean Travis is apparently struggling to keep up with submitted PRs, so I don't have CI result.

I found two issues with your PR, I'm not sure either would have caused this problem, but I changed them in my branch:

1. alphametics has a `jest_0` directory with a few artifacts.  That dir should be removed.
2. `config.json` has a reference to `flatten-array`, which is not part of this track.

Anyway, as soon as I see a result from Travis I'll report back, and we can go from there.

**Comment by matthewmorgan** [comment]
@arthurchipdean I made a new branch from yours [here](https://github.com/exercism/ecmascript/pull/309).

I fixed a few issues (flatten-array is not in master, but was in your branch, removed jest artifacts) and everything is now passing the CI build on Travis.

It looks to me like everything in my branch is what you intended to be in your PR: altered config.json files for all exercises, and the new alphametics exercise.

I would suggest that you pull the main exercism repo down again and merge my branch `acd-resolved-config` in to this one.  Push it back up, let's see if the build passes. I believe it will.

**Comment by matthewmorgan** [comment]
@arthurchipdean I would like to see your work incorporated into the repo.  Let me know if you'd like to finish up this PR, or I can instead merge #309.

Either way, we appreciate your contribution!

**Comment by matthewmorgan** [comment]
Closed in favor of #309 .  @arthurchipdean Thanks for doing the heavy lifting on this one!
